sktime/forecasting/hf_momentfm_forecaster.py

```diff
@@ -0,0 +1,780 @@
+        self.config = config
+        self._config = config if config is not None else {}
+        self.criterion = MSELoss()
+        self._moment_seq_len = 512
+        self.return_model_to_cpu = return_model_to_cpu
+
+    def _fit(self, fh, y, X=None):
+        """Assumes y is a single or multivariate time series."""
+        from torch.optim import Adam
+        from torch.optim.lr_scheduler import OneCycleLR
+        from torch.utils.data import DataLoader
+
+        # initialize accelerator
+        accelerator = Accelerator()
+        # keep a copy of y in case y is None in predict
+        self._y = y
+        self._y_index = self._y.index
+        self._y_cols = self._y.columns
+        self._y_shape = self._y.values.shape
+        self._pretrained_model_name_or_path = (
+            self._config["pretrained_model_name_or_path"]
+            if "pretrained_model_name_or_path" in self._config.keys()
+            else self.pretrained_model_name_or_path
+        )
+        self._freeze_encoder = (
+            self._config["freeze_encoder"]
+            if "freeze_encoder" in self._config.keys()
+            else self.freeze_encoder
+        )
+        self._freeze_embedder = (
+            self._config["freeze_embedder"]
+            if "_freeze_embedder" in self._config.keys()
+            else self.freeze_embedder
+        )
+        self._freeze_head = (
+            self._config["freeze_head"]
+            if "freeze_head" in self._config.keys()
+            else self.freeze_head
+        )
+        self._dropout = (
+            self._config["dropout"]
+            if "dropout" in self._config.keys()
+            else self.dropout
+        )
+        self._head_dropout = (
+            self._config["head_dropout"]
+            if "head_dropout" in self._config.keys()
+            else self.head_dropout
+        )
+        self._transformer_backbone = (
+            self._config["transformer_backbone"]
+            if "transformer_backbone" in self._config.keys()
+            else self.transformer_backbone
+        )
+        self._criterion = (
+            self._config["criterion"]
+            if "criterion" in self._config.keys()
+            else self.criterion
+        )
+        # evaluate the sequence length passed by the user
+        self._seq_len = (
+            self._config["seq_len"]
+            if "seq_len" in self._config.keys()
+            else self.seq_len
+        )
+        if self._seq_len > self._moment_seq_len:
+            warnings.warn(
+                f"length of {self._seq_len} was found which is greater than 512. "
+                "The most recent 512"
+                " values will be used when fitting.",
+                stacklevel=2,
+            )
+            self._seq_len = 512
+        # in the case the config contains 'forecast_horizon', we'll set
+        # fh as that, otherwise we override it using the fh param
+        self._fh_config = (
+            self._config["forecast_horizon"]
```

Comment:
Does MomentFM has the parameter forecast_horizon? 

If yes we should think about which is more important the value from the config or the value the user passes in fit.